### PR TITLE
Provide TimedDiscoveryCache

### LINKED ISSUE #45: Provide a time-based discovery cache ?
I noticed the nonce store is time-based (1 minute lifetime) -- how about also providing a time-based discovery cache ? Am I right that the discovery cache would make changes to the OpenID URI page invisible once the discovered information is cached, at the moment ?

### PR DESCRIPTION
**Pull request by strk**
Fixes #45

### CODE REVIEW

**Comment by strk** [comment]
I'd fix `gofmt` test if I knew how to install `gofmt`...

**Comment by xStrom** [comment]
`gofmt` is included with the default Go installation and is located in the same directory as `go`.

You'll want to run something like `gofmt -l -s -w .\` while being in the source directory.

**Comment by yohcop** [comment]
You can also just look at the couple of lines of diffs in the failed test, and re-adjust those. I think there was 3 lines only.
But anyway, I don't think this should be submitted as is, in this location. See discussion on the bug.

File: discovery_cache.go [hunk 1]
```diff
@@ -67,3 +68,46 @@ func compareDiscoveredInfo(a DiscoveredInfo, opEndpoint, opLocalID, claimedID st
...
+
+	s.cache[id] = timedDiscoveredInfo{info:info, time:time.Now()}
+}
+
+func (s *TimedDiscoveryCache) Get(id string) DiscoveredInfo {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+
+	// Delete old cached while we are at it.
+	newCache := map[string]timedDiscoveredInfo{}
```
Comment (yohcop) [inline discovery_cache.go]: Why not delete(s.cache, k)? Why create a whole new map?

**Comment by yohcop** [comment]
So after discussion on #45 I guess we are not going to merge this. But thanks anyway :)
Like I said I started a short list of such libraries. So if you ever put your code in a separate library, feel free to send a pull request adding a link in the README file!
Thanks!

**Pull request closed without merging**